### ISSUE TITLE
pip install failing on Big Sur with unsupported tag ('cp38', 'cp38', 'macosx_11_0_x86_64')

### ISSUE THREAD

**Issue by matt-dalton** [ionelmc/python-lazy-object-proxy#47]
I'm bringing this issue across from: https://github.com/pypa/wheel/issues/381
This is similar to this closed issue, except I'm using pythong 3.8.2: https://github.com/ionelmc/python-lazy-object-proxy/issues/46

When I pip install this library, I get the following error:
```
File "/private/var/folders/g8/nfhghf95hjgjgghqqb00000gn/T/pip-build-env-cs6vzugw/overlay/lib/python3.8/site-packages/wheel/bdist_wheel.py", line 328, in run
      impl_tag, abi_tag, plat_tag = self.get_tag()
    File "/private/var/folders/g8/nfhghf95hjgjgghqqb00000gn/T/pip-build-env-cs6vzugw/overlay/lib/python3.8/site-packages/wheel/bdist_wheel.py", line 278, in get_tag
assert tag in supported_tags, "would build wheel with unsupported tag {}".format(tag)
  AssertionError: would build wheel with unsupported tag ('cp38', 'cp38', 'macosx_11_0_x86_64')
  ----------------------------------------
  ERROR: Failed building wheel for lazy-object-proxy
```

I was originally getting the same issue in 5 libraries: `lazy-object-proxy`, `pyyaml`, `tornado`, `simplejson`, `psycopg2`) This problem has now been fixed in wheel master, and fixing my wheel version to the latest commit resolves the error in the other 4 libraries. I still get the error in `lazy-object-proxy` though. 

Is there something specific about the way the project uses wheel that would be blocking this fix? Is there a workaround?

Versions are:
- wheel 0.35.1 (fixed to master to use the latest fix)
- pip 20.2.4
- packaging 20.5
- Python 3.8.2

**Comment by ionelmc** [comment]
I think removing the build deps from lazy-object-proxy (rm pyproject.toml should be enough I guess) would prevent the overlay being used. The overlay I believe will use the unpatched release of wheel. But this is a workaorund ofcouse - I can't commit this since pyproject.toml is there for other reasons.

**Comment by matt-dalton** [comment]
Ah yeah - if I fork the library and remove `pyproject.toml` it works fine. Will this at least use the latest version of wheel once the fix is officially released?

**Comment by ionelmc** [comment]
Yes, I'll add a min version pin in pyproject.toml - or you can make a PR if I forget :-)

**Comment by ionelmc** [comment]
Errrr ... min version constraint, not pin.

**Comment by ionelmc** [comment]
Is this still an issue?
